### ISSUE TITLE
Stop using null unnecessarily in source code

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [Microsoft/tslint-microsoft-contrib#490]
There are many places in source code that use `null` and `undefined` interchangeably. As a code cleanup task, it'd be nice to consolidate on only using `undefined` whenever possible. This is what most libraries do, including TSLint and TypeScript.

**Comment by reduckted** [comment]
There's a number of places in the code where it's using `x != null` or `x !== null`. For example: 

https://github.com/Microsoft/tslint-microsoft-contrib/blob/7e87ab24758fef21611d13091e74c55c4fc70a4e/src/chaiPreferContainsToIndexOfRule.ts#L64-L66

When dealing with strings, `x === undefined` would be required to keep the same behavior, but when dealing with objects, which of these is the preferred style?:
```ts
if (x) {
}
```
or
```ts
if (x !== undefined)
```

**Comment by JoshuaKGoldberg** [comment]
> which of these is the preferred style

It'd be nice to eventually also enable `strict-boolean-expressions` (#491), so the `x !== undefined` would be nice in that regard. I've been bitten by `x` equaling `0` enough times to be wary of `if (x)`.

However, if it makes it easier and less prone to bugs _(considering the many places where variables are `* | null | undefined`, `if (x)` should be fine to start here. We can always do another pass-through before enabling `strict-boolean-expressions`.
